app/core/edge_inference.py

```diff
@@ -183,16 +194,17 @@ def inference_is_available(self, detector_id: str) -> bool:
         """
         try:
             inference_client_url = self.inference_client_urls[detector_id]
+            oodd_inference_client_url = self.oodd_inference_client_urls[detector_id]
         except KeyError:
-            logger.debug(f"Failed to look up inference client for {detector_id}")
+            logger.info(f"Failed to look up inference clients for {detector_id}")
             return False
 
-        if not is_edge_inference_ready(inference_client_url):
-            logger.debug("Edge inference server is not ready")
+        if not (is_edge_inference_ready(inference_client_url) and is_edge_inference_ready(oodd_inference_client_url)):
```

Comment:
nit: imo `not A or not B` is a bit easier to reason about than `not (A and B)`

In general one goal of writing code should be to reduce cognitive overhead, and so another strategy is to pull out logic into a boolean variable, like:

```
inference_servers_are_ready = is_edge_inference_ready(inference_client_url) and is_edge_inference_ready(oodd_inference_client_url)
if not inference_servers_are_ready:
	...
```